File: 1503_Last_Moment_Before_All_Ants_Fall_Out_Of_A_Plank.py
class Solution:
    def getLastMoment(self, n: int, left: List[int], right: List[int]) -> int:
        if len(left) == 0:
            totalDistanceLeft = 0
        else:
            totalDistanceLeft = max(left)
        if len(right) == 0:
            totalDistanceRight = n
        else:
            totalDistanceRight = min(right)
        return max(totalDistanceLeft-0, n-totalDistanceRight)

File: 1503_Last_Moment_Before_All_Ants_Fall_Out_Of_A_Plank.java
class Solution {
    public int getLastMoment(int n, int[] left, int[] right) {
        int totalDistanceLeft;
        int totalDistanceRight;
        if (left.length == 0)
            totalDistanceLeft = 0;
        else 
            totalDistanceLeft = Arrays.stream(left).max().getAsInt();
        if (right.length == 0)
            totalDistanceRight = n;
        else 
            totalDistanceRight = Arrays.stream(right).min().getAsInt();
        return Math.max(totalDistanceLeft, n-totalDistanceRight);
        
    }
}
